provider/pkg/provider/provider.go

```diff
@@ -531,6 +614,26 @@ func (k *kubeProvider) Configure(_ context.Context, req *pulumirpc.ConfigureRequ
 			warningConfig := rest.CopyConfig(config)
 			warningConfig.WarningHandler = rest.NoWarnings{}
 			k.config = warningConfig
+			k.kubeconfig = kubeconfig
+
+			namespace := "default"
+			if k.defaultNamespace != "" {
+				namespace = k.defaultNamespace
+			}
+			k.helmReleaseProvider, err = newHelmReleaseProvider(
```

Comment:
I'm not sure how much overhead this causes, but we might want to make this conditional on the program using a helm Release resource.

Reply:
Agreed. Though this should be fairly lightweight for the moment. Happy to revisit later.